sdk/eventhub/Azure.Messaging.EventHubs/assets/functions/azure-principal-functions.ps1

```diff
@@ -0,0 +1,722 @@
+
+#region Principal
+
+function GenerateRandomCredentials()
+{
+  <#
+    .SYNOPSIS
+      It creates random credentials and returns them.
+  #>
+
+  return New-Object Microsoft.Azure.Commands.ActiveDirectory.PSADPasswordCredential -Property @{StartDate=Get-Date; EndDate=Get-Date -Year 2099; Password="$(GenerateRandomPassword)"} 
+}
+
+function GenerateRandomPassword() 
+{
+  <#
+    .SYNOPSIS
+      Generates a random password that can be assigned to a service principal.
+      
+    .DESCRIPTION
+      The password generated by this function will contain a mix of alpha,
+      numeric, and special characters and will vary a small amount in its length.        
+  #>
+
+  $baseLength = (Get-Random -Minimum 28 -Maximum 37)      
+  $upper = (Get-Random -Minimum 6 -Maximum ([int][Math]::Ceiling($baseLength / 3)))  
+  $special = (Get-Random -Minimum 2 -Maximum 3)     
+  $lower = ($baseLength - $upper - $special - $special)                                 
+
+  $password = SelectRandomCharacters $lower "abcdefghiklmnoprstuvwxyz"
+  $password += SelectRandomCharacters $upper "ABCDEFGHKLMNOPRSTUVWXYZ"
+  $password += SelectRandomCharacters $special "1234567890"
+  $password += SelectRandomCharacters $special "!$%&/()=?}][{@#*+"
+
+  $scrambled = ($password.ToCharArray()) | Get-Random -Count ($password.Length)
+  return (-join $scrambled)
+}
+
+function SelectRandomCharacters 
+{    
+  <#
+    .SYNOPSIS
+      Selects a number of random characters from a set.
+  #>
+
+  param
+  (
+    [Parameter(Mandatory=$true)]
+    [int]$length, 
+
+    [Parameter(Mandatory=$true)]
+    [string]$characters
+  ) 
+
+  $random = (1..$length | ForEach { Get-Random -Maximum $characters.length })
+  return (-join $characters[$random])
+}
+
+#endregion Principal
+
+#region EventHubs
+
+function CreateHubIfMissing()
+{
+  <#
+    .SYNOPSIS
+      It tries to retrieve the specified namespace.
+      It creates if it may not be found.
+  #>
+
+  param
+  (
+    [Parameter(Mandatory=$true)]
+    [string] $resourceGroupName,
+
+    [Parameter(Mandatory=$true)]
+    [string] $namespaceName,
+
+    [Parameter(Mandatory=$true)]
+    [string] $eventHubName
+  )
+
+  Write-Host "`t...Requesting eventHub"
+  
+  $eventHub = Get-AzEventHub -ResourceGroupName $resourceGroupName `
+                             -NamespaceName $namespaceName `
+                             -EventHubName $eventHubName `
+                             -ErrorAction SilentlyContinue
+
+  if ($eventHub -eq $null)
+  {
+      # Creates the Event Hub if does not exist
+      Write-Host "`t...Creating new eventHub"
+
+      New-AzEventHub -ResourceGroupName $resourceGroupName `
+                     -NamespaceName $namespaceName `
+                     -EventHubName $eventHubName | Out-Null
+
+      return $true
+  }
+
+  return $false
+}
+
+function GetFullyQualifiedDomainName()
+{
+  <#
+    .SYNOPSIS
+      It takes an access key as input.
+      It returns the fully qualified domain name (FQDN).
+
```

Comment:
In here might be nice to put an example of what an FQDN will look like "{namespacename}.servicebus.windows.net"